packages/core/src/RenderingEngine/vtkClasses/vtkStreamingOpenGLVolumeMapper.js

```diff
@@ -117,16 +131,44 @@ function vtkStreamingOpenGLVolumeMapper(publicAPI, model) {
           oTable
         );
       }
-      model.opacityTextureString = toString;
+      if (scalarOpacityFunc) {
+        model._openGLRenderWindow.setGraphicsResourceForObject(
+          scalarOpacityFunc,
+          model.opacityTexture,
+          toString
+        );
+        if (scalarOpacityFunc !== model._scalarOpacityFunc) {
+          model._openGLRenderWindow.registerGraphicsResourceUser(
+            scalarOpacityFunc,
+            publicAPI
+          );
+          model._openGLRenderWindow.unregisterGraphicsResourceUser(
+            model._scalarOpacityFunc,
+            publicAPI
+          );
+        }
+        model._scalarOpacityFunc = scalarOpacityFunc;
+      }
+    } else {
+      model.opacityTexture = opTex.oglObject;
     }
 
     // rebuild color tfun?
```

Comment:
Is it possible to split some of these commented sections out into their own methods?  I prefer to have a large complex function be something like:

f() {
   doThis();
   doThat();
   doThisOtherTHing();
}

Reply:
Not really, since this is a "fork" of the vtk volume mapper, we should keep things similar to how it is done there